src/modules/notifications/notifications.controller.ts

```diff
@@ -0,0 +1,83 @@
+import { Controller, Get, Req } from '@nestjs/common';
+import { NotificationsService } from './notifications.service';
+import { UserPayload } from '../user/interfaces/user-payload.interface';
+import { ApiBearerAuth, ApiTags, ApiResponse } from '@nestjs/swagger';
+
+@ApiBearerAuth()
+@ApiTags('notifications')
+@Controller('notifications')
+export class NotificationsController {
+  constructor(private readonly notificationsService: NotificationsService) {}
+
+  @Get()
+  @ApiResponse({
+    status: 200,
+    description: 'Notifications retrieved successfully',
+    schema: {
+      example: {
+        status: 'success',
+        status_code: 200,
+        message: 'Notifications retrieved successfully',
+        data: {
+          total_notification_count: 10,
+          total_unread_notification_count: 2,
+          notifications: [
+            {
+              notification_id: 1,
+              is_read: false,
+              message: 'New message received',
+              created_at: '2023-07-30T12:34:56Z',
+            },
+            {
+              notification_id: 2,
+              is_read: true,
+              message: 'Your order has been shipped',
+              created_at: '2023-07-29T11:22:33Z',
+            },
+          ],
+        },
+      },
+    },
```

Comment:
Just add a type of your dto

For example.
```suggestion
type: NotificationResponseSuccessDto
```

Reply:
resolved